src/Symfony/Component/AssetMapper/ImportMap/ImportMapUpdateChecker.php

```diff
@@ -11,16 +11,20 @@
 
 namespace Symfony\Component\AssetMapper\ImportMap;
 
+use Symfony\Component\HttpClient\HttpClient;
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 class ImportMapUpdateChecker
 {
     private const URL_PACKAGE_METADATA = 'https://registry.npmjs.org/%s';
 
+    private readonly HttpClientInterface $httpClient;
+
     public function __construct(
         private readonly ImportMapConfigReader $importMapConfigReader,
-        private readonly HttpClientInterface $httpClient,
+        ?HttpClientInterface $httpClient = null,
     ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
```

Comment:
This was already suggested in #57073 which I considered to be a new feature (see https://github.com/symfony/symfony/pull/57073#discussion_r1611731245).

Reply:
Ok sorry i've lost my way after #57219  :( 

(Maybe stupid idea) Could we, for Symfony 6.4 & 7.1, register a "fake" command (when http_client is not enabeld) that would display a message asking to enable http_client ? 

Or would this be not worth it ? 

As 6.4 is an LTS, i'm afraid of the number of people who could be impacted... 
... but after all there is probably not _that many_ people disabling http_client in framework bundle ?

Reply:
> (Maybe stupid idea) Could we, for Symfony 6.4 & 7.1, register a "fake" command (when http_client is not enabeld) that would display a message asking to enable http_client ?

That's exactly what I was thinking for the Symfony UX Icon search command :)